### PR TITLE
[rtl,clkmgr] Declare clkmgr at D3

### LINKED ISSUE #22482: [clkmgr] D3 Signoff

### PR DESCRIPTION
**Pull request by matutem**
The justification is added as a comment in https://github.com/lowRISC/opentitan/issues/22482.

Fixes #22482

### CODE REVIEW

**Comment by matutem** [comment]
CHANGE AUTHORIZED: hw/ip_templates/clkmgr/data/clkmgr.hjson.tpl

**Review by vogelpi — COMMENTED** [review]
Thanks @matutem for doing the signoff work. The PR and the changes to the checklist look good to me. But for documentation purposes we would need to populate the signoff issue and report the findings made during the signoff. Would you be able to do this please?

To get an idea of what is needed, please refer to the `lc_ctrl` D3 signoff issue: https://github.com/lowRISC/opentitan/issues/22480#issuecomment-2254153756

Specifically referring to the lint checklist item: we need to check whether clkmgr was at D3 before and if yes, if the lint waiver file for AscentLint has changed since then. If it hasn't been at D3 before, or if the waiver file changed in the meantime, we need to argue why these waiver rules are required and TC needs to approve this.

**Comment by matutem** [comment]
@vogelpi I updated https://github.com/lowRISC/opentitan/issues/22482 with the justification for the signoff. The block was not previously at D3, but I added some explanation for the lint waivers. Let me know if you think it needs more details.

**Comment by matutem** [comment]
This also needs another `CHANGE AUTHORIZED` comment to get through.

File: hw/ip_templates/clkmgr/doc/checklist.md [hunk 1]
```diff
@@ -100,17 +100,17 @@ Security      | [SEC_CM_COUNCIL_REVIEWED][]  | Done        |
...
-Code Quality  | [RDC_COMPLETE][]        | Not Started |
-Review        | [REVIEW_RTL][]          | Not Started |
-Review        | [REVIEW_DELETED_FF][]   | Not Started |
-Review        | [REVIEW_SW_CHANGE][]    | Not Started |
-Review        | [REVIEW_SW_ERRATA][]    | Not Started |
-Review        | Reviewer(s)             | Not Started |
-Review        | Signoff date            | Not Started |
+Documentation | [NEW_FEATURES_D3][]     | Done        |
+RTL           | [TODO_COMPLETE][]       | Done        |
+Code Quality  | [LINT_COMPLETE][]       | Done        |
```
Comment (andreaskurth) [inline hw/ip_templates/clkmgr/doc/checklist.md:105]:
```suggestion
Code Quality  | [LINT_COMPLETE][]       | Done        | With waivers approved by TC on 2024-08-08
```
Please stage this change for now. I'll add it to the agenda of the TC meeting and report the result back here.

File: hw/ip_templates/clkmgr/doc/checklist.md [hunk 2]
```diff
@@ -100,17 +100,17 @@ Security      | [SEC_CM_COUNCIL_REVIEWED][]  | Done        |
...
+Code Quality  | [CDC_COMPLETE][]        | Waived      | No block-level flow available - waived to top-level signoff.
+Code Quality  | [RDC_COMPLETE][]        | Waived      | No block-level flow available - waived to top-level signoff.
+Review        | [REVIEW_RTL][]          | Done        |
+Review        | [REVIEW_DELETED_FF][]   | Done        |
+Review        | [REVIEW_SW_CHANGE][]    | Done        |
+Review        | [REVIEW_SW_ERRATA][]    | Done        |
+Review        | Reviewer(s)             | Done        | @matutem, @vogelpi
```
Comment (andreaskurth) [inline hw/ip_templates/clkmgr/doc/checklist.md:112]: Please add me (`adk@`) as a reviewer

Comment (andreaskurth) [reply hw/ip_templates/clkmgr/doc/checklist.md:112]: Oh, and I think these are OpenTitan handles and should be written `name@` (i.e., `@` at the end) in order to not have GitHub confuse them with GitHub account names

**Comment by andreaskurth** [comment]
CHANGE AUTHORIZED: hw/ip_templates/clkmgr/data/clkmgr.hjson.tpl
CHANGE AUTHORIZED: hw/top_earlgrey/ip_autogen/clkmgr/data/clkmgr.hjson

Only `design_stage` changed, which doesn't impact RTL

**Comment by matutem** [comment]
@andreaskurth I made the suggested changes. Thanks.

**Review by andreaskurth — APPROVED** [review]
Looks all good to me now. Thanks @matutem!

**Pull request merged**